### PR TITLE
WIP: Storage + Rocket League + AR + ACO

### PR DESCRIPTION
**Pull request by osztenkurden**

### CODE REVIEW

File: server/api/bakkesmod.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	if (fs.existsSync(bakkesModConfigPath)) status.bakkesModStartedOnce = true;
+	if (fs.existsSync(bakkesModPath)) status.bakkesModInstalled = true;
+	if (fs.existsSync(path.join(bakkesModDirPath, sosPluginFiles[0]))) status.sosPluginInstalled = true;
+	if (verifyPluginList()) status.sosConfigSet = true;
+
+	return res.json({ success: true, status });
+};
+
+export const downloadBakkesMod: express.RequestHandler = async (req, res) => {
+	request(bakkesModDownloadUrl)
```
Comment (osztenkurden) [inline server/api/bakkesmod.ts:90]: Remove request - use fetch

File: server/api/bakkesmod.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+			return res.json({ success: false });
+		})
+		.on('end', () => {
+			return res.json({ success: true, path: bakkesModDownloadFilePath });
+		})
+		.pipe(fs.createWriteStream(bakkesModDownloadFilePath));
+};
+
+export const downloadSosPlugin: express.RequestHandler = async (req, res) => {
+	request(sosPluginDownloadAPIPath, (error, _response, body) => {
```
Comment (osztenkurden) [inline server/api/bakkesmod.ts:101]: Remove request - use fetch #2

File: server/api/bakkesmod.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+		return res.json({ success: true });
+	});
+};
+
+export const installSosPlugin: express.RequestHandler = async (req, res) => {
+	if (!fs.existsSync(sosPluginDownloadFilePath))
+		return res.json({ success: false, message: 'SOS plugin needs to be downloaded first' });
+
+	if (fs.existsSync(sosPluginExtractPath)) {
+		await del(sosPluginExtractPath, { force: true, expandDirectories: true });
```
Comment (osztenkurden) [inline server/api/bakkesmod.ts:142]: Can't we use fs.unlinkSync(path, { recursive: true })?

File: server/api/bakkesmod.ts (hunk 1 above)
Comment (osztenkurden) [reply server/api/bakkesmod.ts:90]: @kherchel

File: server/api/bakkesmod.ts (hunk 3 above)
Comment (osztenkurden) [reply server/api/bakkesmod.ts:179]: Nvm, rmSync is available from Node 14, Electron 11 is on Node 12.

Comment (kherchel) [reply server/api/bakkesmod.ts:174]: nah, doesn't work / glitches out on non-empty directories

**Pull request merged**